### PR TITLE
[android] Kotlinize scoped universal modules

### PR DESCRIPTION
**Pull request by wschurman**
# Why

Part 17 of #13685.

# How

1. Convert automatically
1. Clean up manually
1. Verify line-by-line with original java to ensure correct conversion

# Test Plan

load experience

### CODE REVIEW

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFirebaseCoreService.kt [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      } catch (err: Exception) {
+        null
+      }
+    }
+
+    private fun MutableMap<String, String>.addJSONStringToMap(
+      json: JSONObject?,
+      path: String,
+      name: String
+    ) {
```
Comment (ide) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFirebaseCoreService.kt:101]: Lines up better with MutableMap's other APIs

```suggestion
    private fun MutableMap<String, String>.putJSONString(
      key: String,
      json: JSONObject?,
      path: String
    ) {
```

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFirebaseCoreService.kt [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    val defaultAppName = defaultApp?.name ?: DEFAULT_APP_NAME
+
+    // Get experience key & unique app name
+    appName = "__sandbox_" + getEncodedExperienceScopeKey(experienceKey)
+    appOptions = getOptionsFromManifest(manifest)
+
+    // Add the app to the list of protected app names
+    synchronized(protectedAppNames) {
+      protectedAppNames[defaultAppName] = false
+      protectedAppNames.put(appName, false)
```
Comment (ide) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFirebaseCoreService.kt:186]:
```suggestion
      protectedAppNames[appName] = false
```

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedSecureStoreModule.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  private val scopedSharedPreferences: SharedPreferences
+    get() = scopedContext.getSharedPreferences(
+      SHARED_PREFERENCES_NAME,
+      Context.MODE_PRIVATE
+    )
+
+  companion object {
+    private const val SHARED_PREFERENCES_NAME = "SecureStore"
```
Comment (ide) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedSecureStoreModule.kt:39]: Module scope variable

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/UpdatesBinding.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import host.exp.exponent.kernel.KernelProvider
+import host.exp.exponent.di.NativeModuleDepsProvider
+import host.exp.exponent.kernel.KernelConstants
+import java.io.File
+
+class UpdatesBinding(context: Context, experienceProperties: Map<String, Any?>) :
+  UpdatesService(context), UpdatesInterface {
+
+  @Inject
+  lateinit var mDatabaseHolder: DatabaseHolder
```
Comment (ide) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/UpdatesBinding.kt:25]:
```suggestion
  lateinit var databaseHolder: DatabaseHolder
```

**Review by ide — APPROVED** [review]

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ConstantsBinding.kt [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      val metrics = resources.displayMetrics
+      val dp = px / (metrics.densityDpi / 160f)
+      return dp.toInt()
+    }
+  }
+
+  init {
+    NativeModuleDepsProvider.instance.inject(ConstantsBinding::class.java, this)
+    val resourceId = context.resources.getIdentifier("status_bar_height", "dimen", "android")
+    mStatusBarHeight = if (resourceId > 0) convertPixelsToDp(
```
[the remark this answers is not in the document]
Comment (wschurman) [reply android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ConstantsBinding.kt:90]: Yep!

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFileSystemModule.kt [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      this["bundledAssets"] = getBundledAssets()
+    }
+  }
+
+  private fun getBundledAssets(): List<String>? {
+    return if (!Constants.isStandaloneApp()) {
+      // Fastpath, only standalone apps support bundled assets.
+      null
+    } else try {
+      val inputStream = context.assets.open(SHELL_APP_EMBEDDED_MANIFEST_PATH)
```
[the remark this answers is not in the document]
Comment (wschurman) [reply android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFileSystemModule.kt:22]: Changed to `use`.

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/UpdatesBinding.kt (hunk 4 above)
Comment (wschurman) [reply android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/UpdatesBinding.kt:25]: Needs to be something other than databaseHolder not to conflict, I chose `databaseHolderInternal`.

**Pull request merged**